HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m

```diff
@@ -691,20 +686,156 @@ - (void)testConcurrentTracking {
-    [self tearDownMixpanel];
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        [self setUpMixpanel];
+        Mixpanel *testMixpanel = [[Mixpanel alloc] initWithToken:[self randomTokenId] andFlushInterval:60];
+        XCTAssertNotNil(testMixpanel);
         [expectation fulfill];
     });
-    [self waitForExpectationsWithTimeout:5 handler:nil];
+    [self waitForExpectationsWithTimeout:2 handler:nil];
 }
 
+- (void)testMPDB {
+    NSUInteger randomId = arc4random();
+    NSString *testToken = [NSString stringWithFormat:@"%lu", (unsigned long)randomId];
+    int numRows = 50;
+    int halfRows = numRows/2;
+    NSString *eventName = @"Test Event";
+    [self removeDBfile:testToken];
+    MPDB *mpdb = [[MPDB alloc] initWithToken:testToken];
+    [mpdb open];
+    for (NSString *pType in @[@"events", @"people", @"groups"]) {
+        NSArray *emptyArray = [mpdb readRows:pType numRows:numRows flag:false];
+        XCTAssertTrue(emptyArray.count == 0, "Table should be empty");
+        for (int i = 0; i < numRows; i++) {
+            NSDictionary *eventObj = @{@"event": eventName, @"properties": @{@"index": @(i)}};
+            NSData *eventData = [NSJSONSerialization dataWithJSONObject:eventObj options:0 error:NULL];
+            [mpdb insertRow:pType data:eventData flag:false];
+            NSLog(@"%d", i);
+        }
+        NSArray *dataArray = [mpdb readRows:pType numRows:numRows flag:halfRows];
+        NSMutableArray *ids = [[NSMutableArray alloc] init];
+        NSUInteger index = 0;
+        for (NSDictionary *entity in dataArray) {
+            [ids addObject:entity[@"id"]];
+            XCTAssertEqual(entity[@"event"], eventName, "Event name should be unchanged");
+            // index should be oldest events, 0 - 24
+            XCTAssertEqual(entity[@"properties"], @{@"index": @(index)}, "Should read oldest events first");
+            index++;
+        }
+
+        [mpdb deleteRows:pType ids:@[@1, @2, @3]];
+        NSArray *dataArray2 = [mpdb readRows:pType numRows:numRows flag:false];
+        // even though we request numRows, there should only be numRows - 3 left
+        XCTAssertEqual([dataArray2 count], 47);
+        NSUInteger index2 = 0;
+        for (NSDictionary *entity in dataArray2) {
+            XCTAssertEqualObjects(entity[@"event"], eventName, "Event name should be unchanged");
+            // oldest events (0-2) should have been deleted so index should be more recent events 3-49
+            NSNumber *idx = entity[@"properties"][@"index"];
+            XCTAssertEqualObjects(idx, @(index2 + 3), "Should have deleted oldest events first");
+            index2++;
+        }
+    }
+    [mpdb close];
+    // TODO: What's up with this log we get when deleting the db file? Connection should be closed and nil.
+    // BUG IN CLIENT OF libsqlite3.dylib: database integrity compromised by API violation: vnode unlinked while in use
```

Comment:
did you encounter this? I'm not sure if it's actually a problem or just an annoyance.